app/celery/letters_pdf_tasks.py

```diff
@@ -56,62 +50,63 @@
-                reference=str(notification.reference), billable_units=billable_units))
+        encrypted_data = encryption.encrypt(letter_data)
 
-    except (RequestException, BotoClientError):
+        notify_celery.send_task(
+            name=TaskNames.CREATE_LETTER_PDF,
+            args=(encrypted_data,),
+            queue=QueueNames.SANITISE_LETTERS
+        )
+    except Exception:
         try:
             current_app.logger.exception(
-                "Letters PDF notification creation for id: {} failed".format(notification_id)
+                "RETRY: calling create-letter-pdf task for notification {} failed".format(notification_id)
             )
             self.retry(queue=QueueNames.RETRY)
-        except MaxRetriesExceededError:
-            current_app.logger.error(
-                "RETRY FAILED: task create_letters_pdf failed for notification {}".format(notification_id),
-            )
-            update_notification_status_by_id(notification_id, 'technical-failure')
-
-
-def get_letters_pdf(template, contact_block, filename, values):
-    template_for_letter_print = {
-        "subject": template.subject,
-        "content": template.content,
-        "template_type": template.template_type
-    }
-
-    data = {
-        'letter_contact_block': contact_block,
-        'template': template_for_letter_print,
-        'values': values,
-        'filename': filename,
-    }
-    resp = requests_post(
-        '{}/print.pdf'.format(
-            current_app.config['TEMPLATE_PREVIEW_API_HOST']
-        ),
-        json=data,
-        headers={'Authorization': 'Token {}'.format(current_app.config['TEMPLATE_PREVIEW_API_KEY'])}
-    )
-    resp.raise_for_status()
+        except self.MaxRetriesExceededError:
+            message = "RETRY FAILED: Max retries reached. " \
+                      "The task create-letter-pdf failed for notification {}. " \
+                      "Notification has been updated to technical-failure".format(notification_id)
+            update_notification_status_by_id(notification_id, NOTIFICATION_TECHNICAL_FAILURE)
+            raise NotificationTechnicalFailureException(message)
+
+
+@notify_celery.task(bind=True, name="update-billable-units-for-letter", max_retries=15, default_retry_delay=300)
+@statsd(namespace="tasks")
+def update_billable_units_for_letter(self, notification_id, page_count):
+    notification = get_notification_by_id(notification_id, _raise=True)
+
+    billable_units = get_billable_units_for_letter_page_count(page_count)
 
-    pages_per_sheet = 2
-    billable_units = math.ceil(int(resp.headers.get("X-pdf-page-count", 0)) / pages_per_sheet)
+    if notification.key_type != KEY_TYPE_TEST:
+        notification.billable_units = billable_units
+        dao_update_notification(notification)
 
-    return resp.content, billable_units
+    current_app.logger.info(
+        'Letter notification reference {reference}: billable units set to {billable_units}'.format(
```

Comment:
Can you include the notification_id in the log message, please? It's nice to search by id in the Kibanna

Reply:
good suggestion, I don't know how I missed that, I will do that now 👍